lib/middleware/logging_context.rb

```diff
@@ -0,0 +1,29 @@
+require 'securerandom'
+
+module Middleware
+  class LoggingContext
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      session = env['rack.session']
+      ::Logging.mdc['remote_ip'] = env['action_dispatch.remote_ip'].try(:to_s)
+      ::Logging.mdc['request'] = env['action_dispatch.request_id']
+
+      if env["rack.session"].id.present?
+        # use random token to prevent hijack.
+        session['logging_token'] ||= SecureRandom.uuid
```

Comment:
There is a reason - the stack is unable to remove keys from MDC context, it simply logs everything. We would log the session id twice. Since this is a big change in logging which will eventually lead to changes of config if users want to enable journald/syslog I thought it's good time for a change and cleanup.